enumerable.rb

```diff
@@ -0,0 +1,350 @@
+# rubocop:disable all
+module Enumerable
+  def my_each
+    unless block_given?
+      to_enum(__method__)
+    else
+      i = 0
+      while i < self.size
+        yield self[i]
+        i += 1
+      end
+      self
+    end
+  end
+
+  def my_each_with_index
+    unless block_given?
+      to_enum(__method__)
+    else
+      i = 0
+      while i < self.size
+        yield(self[i], i)
+        i += 1
+      end
+      self
+    end
+  end
+
+  def my_select
+    unless block_given?
+      to_enum(__method__)
+    else
+      i = 0
+      arr = []
+      while i < self.size
+        yield self[i]
+        if yield self[i]
+          arr << self[i]
+        end
+        i += 1
+      end
+      arr
+    end
+  end
+
+  def my_all?(word = nil)
+    unless block_given?
+      i = 0
+      j = 0
+
+      while i < self.size
+        self[i]
+        i += 1
+      end
+      if self.size == 0
+        true
+      elsif word.class == Regexp
+
+        for element in self
+          if element.match(word)
+            j += 1
+          end
+        end
+        j == self.length ? true : false
+
+      elsif word == Numeric
+
+        for element in self
+          if element.class <= word
+            j += 1
+          end
+        end
+        j == self.length ? true : false
+      elsif word != nil
+        i = 0
+        while i < self.size
+          if self[i - 1] == self[i]
+            j += 1
+          end
+          i += 1
+        end
+
+        for element in self
+          if element.class == word
+            j += 1
+          end
+        end
+        j == self.length ? true : false
+
+      else
+        for element in self do
+          if element == false or element == nil
+            k += 1
+          end
+        end
+        k == 0 ? true : false
+      end
+    else
+      i = 0
+      j = 0
+      while i < self.size
+        yield self[i]
+        i += 1
+      end
+
+      for element in self
+
+        if yield(element) == false or yield(element) == nil
+          j += 1
+        end
+      end
+      j == 0 ? true : false
+
+    end
+  end
```

Comment:
#### my_all

- [ ] when no block or argument is given returns true only when none of the collection members are false or nil if the mentioned condition is not met it should return false

     > this [nil, false, true, []].my_all? should be equal to this [nil, false, true, []].all? # false

- [ ] when a class is passed as an argument returns true only if all of the collection is a member of such class, otherwise, it should return false

     > [1, 2.5, 'a', 9].my_all?(Integer) should be equal to this [1, 2.5, 'a', 9].all?(Integer) # false